From 073f4d44e50b501afa03e6fdcda6c179f2cbb4ae Mon Sep 17 00:00:00 2001
From: Alex Johnston <acjohnston@google.com>
Date: Mon, 29 Nov 2021 16:06:38 +0000
Subject: [PATCH] Add non system overlay flag to InstallCaCertificateWarning

Bug: 196969991
Test: Manual testing with Settings
Change-Id: Ia9dc251c853526b2ce66c9f8ff595d496b7f1bc4
Merged-In: Ia9dc251c853526b2ce66c9f8ff595d496b7f1bc4
(cherry picked from commit 2f193c81576e2e86cf64470aeb312cab1e76d40d)
(cherry picked from commit fca5cb37486fb1072d6233670b2fd66e555b07c1)
Merged-In:Ia9dc251c853526b2ce66c9f8ff595d496b7f1bc4
---
 .../settings/security/InstallCaCertificateWarning.java      | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/com/android/settings/security/InstallCaCertificateWarning.java b/src/com/android/settings/security/InstallCaCertificateWarning.java
index 91faae1f6a..b5cc6b35c7 100644
--- a/src/com/android/settings/security/InstallCaCertificateWarning.java
+++ b/src/com/android/settings/security/InstallCaCertificateWarning.java
@@ -16,6 +16,8 @@
 
 package com.android.settings.security;
 
+import static android.view.WindowManager.LayoutParams.SYSTEM_FLAG_HIDE_NON_SYSTEM_OVERLAY_WINDOWS;
+
 import android.annotation.Nullable;
 import android.app.Activity;
 import android.content.Intent;
@@ -41,6 +43,8 @@ public class InstallCaCertificateWarning extends Activity {
         super.onCreate(savedInstanceState);
 
         setContentView(R.layout.ca_certificate_warning_dialog);
+        getWindow().addSystemFlags(SYSTEM_FLAG_HIDE_NON_SYSTEM_OVERLAY_WINDOWS);
+
         final GlifLayout layout = findViewById(R.id.setup_wizard_layout);
 
         final FooterBarMixin mixin = layout.getMixin(FooterBarMixin.class);
@@ -52,6 +56,7 @@ public class InstallCaCertificateWarning extends Activity {
                         .setTheme(R.style.SudGlifButton_Secondary)
                         .build()
         );
+        mixin.getSecondaryButtonView().setFilterTouchesWhenObscured(true);
 
         mixin.setPrimaryButton(
                 new FooterButton.Builder(this)
@@ -61,6 +66,7 @@ public class InstallCaCertificateWarning extends Activity {
                         .setTheme(R.style.SudGlifButton_Primary)
                         .build()
         );
+        mixin.getPrimaryButtonView().setFilterTouchesWhenObscured(true);
     }
 
     private View.OnClickListener installCaCertificate() {
-- 
2.17.1

